MountList.lua

```diff
@@ -191,6 +191,37 @@ function LM.MountList:PriorityRandom(r)
 end
 
 function LM.MountList:RarityRandom(r)
+    if #self == 0 then return end
+    if not MountsRarityAddon or not MountsRarityAddon.MountsRarity then
+        LM.Debug(" - RarityRandom: MountsRarity not loaded. Defaulting to SimpleRandom.")
+        return self:SimpleRandom(r)
+    end
+
+    local mountsRarity = {}
+    local totalWeight = 0
+
+    for _, m in ipairs(self) do
+        local rarity = MountsRarityAddon.MountsRarity[tostring(m.mountID)]
+        if not rarity then
+            LM.Debug(format(" - Missing rarity data for mountID %d", m.mountID) .. " (" .. m.name .. ")")
+            rarity = 50.0 -- for unknown mounts
+        end
+        local invertedRarity = ( 1 / rarity )
+
+        mountsRarity[m.mountID] = invertedRarity
```

Comment:
Thinking about it more, this is probably the place where we should also check for priority `0` (aka "disabled").
Makes even more sense to unify the two implementations.

Reply:
Yeah you can just set invertedRarity = 0 if m:GetPriority() == 0 I think.

Reply:
When the priority is set to `0`, it will not be in the list. So this is actually not needed.